aiidalab_widgets_base/viewers.py

```diff
@@ -1175,18 +1195,31 @@ def _observe_displayed_structure(self, change):
                     default_representation=False,
                     name="Structure",
                 )
-                nglview_params = [
-                    rep.nglview_parameters(
-                        np.where(rep.atoms_in_representaion(self.displayed_structure))[
-                            0
-                        ]
-                    )
-                    for rep in self._all_representations
-                    if rep.show.value
-                ]
+                bonds = []
+                nglview_params = []
+                for representation in self._all_representations:
+                    if representation.show.value:
+                        indices = np.where(
+                            representation.atoms_in_representaion(
+                                self.displayed_structure
+                            )
+                        )[0]
+                        nglview_params.append(
+                            representation.nglview_parameters(indices)
+                        )
+
+                        # Add bonds if ball+stick representation is used.
+                        if representation.type.value == "ball+stick":
+                            bonds += self._compute_bonds(
+                                self.displayed_structure[indices],
+                                representation.size.value * 0.2,
+                                representation.color.value,
+                            )
                 self._viewer.set_representations(nglview_params, component=0)
                 self._viewer.add_unitcell()
+                self._viewer._add_shape(set(bonds), name="bonds")
```

Comment:
Even though it is not obvious in the GUI, but you add new shapes for the bonds, and haven't removed the default bond from the nglview itself, thus the two bonds overlap.

Reply:
That was a hard nut to crash, but I think I managed in https://github.com/aiidalab/aiidalab-widgets-base/pull/535/commits/066aa4a5a567b79c726b04d0a002b2b64e5f6792. What do you think @cpignedoli?

Reply:
After setting different radii for different atoms in this commit, the wrong `intermediate_point` becomes obvious. Before, all atoms have the same radius in the `ball+stick` model, thus `intermediate_point` is  (only) correct in that case.

Reply:
I am not sure we do not remove default bonds: we never use the ngl representation "ball+stik" we always use spacefill (no bonds) and add bond shapes in the case we use the "fake ball+stik"

Reply:
v1 + bond_vector * Radius[atom1.symbol] / (
                    Radius[atom1.symbol] + Radius[atom2.symbol]
bond_vector goes from atom1 to atom2.
we will add to atom1 a vector bond_v*r1/(r1+r2) and to atom2 a vector, in opposite direction (or from intermediate point to atom2), bond_v*r2/(r1+r2) in total: bond_v*(r1+r2)/(r1+r2) = bond_v and the two portions of bonds will have length proportional to the size of the spheres